pilot/pkg/bootstrap/server.go

```diff
@@ -218,14 +218,15 @@ func NewServer(args *PilotArgs) (*Server, error) {
 
 	prometheus.EnableHandlingTimeHistogram()
 
-	// TODO: revert to watching k8s (and merge with the file)
 	s.initMeshConfiguration(args, s.fileWatcher)
 
 	// Apply the arguments to the configuration.
 	if err := s.initKubeClient(args); err != nil {
 		return nil, fmt.Errorf("error initializing kube client: %v", err)
 	}
 
+	s.watchMeshConfiguration(args)
```

Comment:
This should be in initMeshConfiguration and only watch it when local file not exist.

They are independent

Reply:
I tried this originally but it didn't work; `watchMeshConfiguration` needs `s.kubeClient` to be set, which is done in `initKubeClient`, and `initKubeClient` needs some mesh config (like the default one) to know whether to initialize `s.kubeClient`. Maybe renaming it to `maybeWatchMeshConfiguration`?